From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: dbus: fix xml parser when used with libxml2

When built with expat, dbus builds fine. However, in combination with
libxml2 we get the following error:

	xml.c:252: undefined reference to `bus_config_parser_check_doctype'
	collect2: ld returned 1 exit status
	make[3]: *** [dbus-daemon-launch-helper] Error 1

I'm not really sure if this is the right fix, but it makes it at least
compile :-) Looks like some of the tools don't use config-parser as they
should.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 bus/Makefile.am |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

Index: dbus-1.2.1/bus/Makefile.am
===================================================================
--- dbus-1.2.1.orig/bus/Makefile.am
+++ dbus-1.2.1/bus/Makefile.am
@@ -85,14 +85,18 @@ dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @
 
 LAUNCH_HELPER_SOURCES=				\
 	$(XML_SOURCES)				\
+	config-parser.c				\
+	config-parser.h				\
 	config-parser-common.c			\
 	config-parser-common.h			\
-	config-parser-trivial.c			\
-	config-parser-trivial.h			\
 	desktop-file.c				\
 	desktop-file.h				\
 	utils.c					\
 	utils.h					\
+	policy.c				\
+	policy.h				\
+	selinux.h				\
+	selinux.c				\
 	activation-exit-codes.h			\
 	activation-helper.h	  		\
 	activation-helper.c
@@ -153,12 +157,16 @@ noinst_PROGRAMS=$(TESTS) dbus-daemon dbu
 
 bus_test_system_SOURCES=			\
 	$(XML_SOURCES)				\
+	config-parser.c				\
+	config-parser.h				\
 	config-parser-common.c			\
 	config-parser-common.h			\
-	config-parser-trivial.c			\
-	config-parser-trivial.h			\
 	utils.c					\
 	utils.h					\
+	policy.c				\
+	policy.h				\
+	selinux.h				\
+	selinux.c				\
 	test-system.c
 
 bus_test_system_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_BUS_LIBS)
